### PR TITLE
feat: add EsBadge component

### PR DESCRIPTION
**Pull request by mpleroux**
### 🔗 Linked issue

[CED-1738: Create v3 EsBadge](https://energysage.atlassian.net/browse/CED-1738)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- New `EsBadge` component based on the previous ES-DS 2.0 version
- Implemented as simple `span` element with badge variant classes, as suggested by Jira ticket

### 🥼 Testing

- Testing locally at http://localhost:8500/molecules/badge

#### 🧐 Feedback Requested / Focus Areas

- Is this implementation _too_ simple? Or all that's needed?
- For the examples, should I ditch the old legacy color scheme of `primary`, `success`, `warning`, etc.?
    - I suppose I could use one shade of the Success, Warning, and Error [Feedback colors](http://localhost:8500/atoms/color).

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.
- [ ] I have documented testing approach

### CODE REVIEW

**Comment by nathanielwarner** [comment]
👀

File: es-ds-docs/pages/molecules/badge.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<template>
+    <div>
+        <h1>Badge</h1>
+        <p>Extended from <a href="https://v3.primevue.org/badge/" target="_blank"> PrimeVue Badge </a></p>
```
Comment (nathanielwarner) [inline es-ds-docs/pages/molecules/badge.vue:4]: It's actually not! You could say "based on Bootstrap badge" instead

**Review by nathanielwarner — COMMENTED** [review]
Generally looks good! re. whether it's too simple... I think simple is better. But we may want to at least implement the "pill" version with the ["pill" prop](https://bootstrap-vue.org/docs/components/badge#pill-badges) ([corresponding bootstrap class](https://getbootstrap.com/docs/4.6/components/badge/#pill-badges)). A [search in github](https://github.com/search?q=org%3AEnergySage+%3Ces-badge&type=code) didn't turn up other parts of `b-badge` that this implicitly relies on.

**Comment by mpleroux** [comment]
I updated the description on the docs page.

After discussing the Bootstrap `pill` prop yesterday we concluded there's no clear advantage to implementing it right now. The default badge styling already uses large rounded corners, and manually adding the `badge-pill` class actually *reduces* the default padding. The `pill` prop is used in only one file in `es-site`, and nowhere else.

**Review by nathanielwarner — APPROVED** [review]
LGTM! 🚀

**Pull request merged**